app/script/util/DebugUtil.js

```diff
@@ -262,4 +262,30 @@ z.util.DebugUtil = class DebugUtil {
     this.logger.log(`-- IndexedDB open: ${this.storageRepository.storageService.db.isOpen()}`);
     this.logger.log(`-- WebSocket ready state: ${window.wire.app.service.web_socket.socket.readyState}`);
   }
+
+  reprocessNotificationStream(conversationId = this.conversationRepository.active_conversation().id) {
+    const clientId = wire.app.repository.client.currentClient().id;
+
+    return this.eventRepository.notificationService
+      .getNotifications(clientId, undefined, z.event.EventRepository.CONFIG.NOTIFICATION_BATCHES.MAX)
+      .then(({notifications}) => {
+        this.logger.info(`Fetched "${notifications.length}" notifications for client "${clientId}".`, notifications);
+
+        const isOTRMessage = notification => notification.type === z.event.Backend.CONVERSATION.OTR_MESSAGE_ADD;
+        const isInCurrentConversation = notification => notification.conversation === conversationId;
+
+        return notifications
+          .map(notification => notification.payload)
+          .reduce((accumulator, payload) => accumulator.concat(payload))
+          .filter(notification => {
+            return isOTRMessage(notification) && isInCurrentConversation(notification);
+          });
+      })
+      .then(events => {
+        this.logger.info(`Reprocessing "${events.length}" OTR messages...`);
+        for (const event of events) {
+          this.eventRepository._processEvent(event, z.event.EventRepository.SOURCE.STREAM);
+        }
```

Comment:
Is there a particular reason for the preference for a `for. .. of` loop over `Array.forEach()`